tests/Aspire.Workload.Tests/WorkloadTestsBase.cs

```diff
@@ -0,0 +1,263 @@
+
+                // endpointsFound: ["foo", "https://localhost:7589/weatherforecast"]
+                foreach (var endpointFound in endpointsFound)
+                {
+                    // matchedEndpoints: ["https://localhost:7589/weatherforecast"]
+                    string[] matchedEndpoints = expectedEndpoints.Where(e => Regex.IsMatch(endpointFound, e)).ToArray();
+                    if (matchedEndpoints.Length == 0)
+                    {
+                        Assert.Fail($"Unexpected endpoint found: {endpointFound} for resource named {resourceName}. Expected endpoints: {string.Join(',', expectedEndpoints)}");
+                    }
+                    matchingEndpoints++;
+                }
+
+                AssertEqual(expectedEndpoints.Length, matchingEndpoints, $"Expected number of endpoints for {resourceName}");
+
+                // Check 'Source' column
+                AssertEqual(expectedRow.Source, await cellLocs[4].InnerTextAsync(), $"Source for {resourceName}");
+
+                foundRows.Add(expectedRow with { Endpoints = endpointsFound });
+                foundNames.Add(resourceName);
+            }
+        }
+
+        if (foundNames.Count != expectedRowsTable.Count)
+        {
+            Assert.Fail($"Expected rows not found: {string.Join(", ", expectedRowsTable.Keys.Except(foundNames))}");
+        }
+
+        return foundRows.ToArray();
+    }
+
+    // Don't fixup the prefix so it can have characters meant for testing, like spaces
+    public static string GetNewProjectId(string? prefix = null)
+        => (prefix is null ? "" : $"{prefix}_") + FixupSymbolName(Path.GetRandomFileName());
+
+    public static async Task<CommandResult?> AssertTestProjectRunAsync(string testProjectDirectory, string testType, ITestOutputHelper testOutput, string config = "Debug", int testRunTimeoutSecs = 3 * 60)
+    {
+        if (testType == "none")
+        {
+            Assert.False(Directory.Exists(testProjectDirectory), "Expected no tests project to be created");
+            return null;
+        }
+        else
+        {
+            Assert.True(Directory.Exists(testProjectDirectory), $"Expected tests project at {testProjectDirectory}");
+            using var cmd = new DotNetCommand(testOutput, label: $"test-{testType}")
+                                    .WithWorkingDirectory(testProjectDirectory)
+                                    .WithTimeout(TimeSpan.FromSeconds(testRunTimeoutSecs));
+
+            var res = (await cmd.ExecuteAsync($"test -c {config}"))
+                                .EnsureSuccessful();
+
+            Assert.Matches("Passed! * - Failed: *0, Passed: *1, Skipped: *0, Total: *1", res.Output);
+            return res;
+        }
+    }
+
+    internal static async Task AssertStarterTemplateRunAsync(IBrowserContext context, AspireProject project, string config, ITestOutputHelper _testOutput)
```

Comment:
This is a bit odd to have in the WorkloadTestBase class. It seems like it is specific to certain tests.